packages/flutter_tools/test/general.shard/build_system/targets/dart_test.dart

```diff
@@ -46,434 +42,443 @@ void main() {
-      packagesPath: anyNamed('packagesPath'),
-      fileSystemRoots: anyNamed('fileSystemRoots'),
-      fileSystemScheme: anyNamed('fileSystemScheme'),
-      platformDill: anyNamed('platformDill'),
-      initializeFromDill: anyNamed('initializeFromDill'),
-      dartDefines: anyNamed('dartDefines'),
-    )).thenAnswer((Invocation invocation) async {
-      return null;
-    });
-    final BuildResult result = await buildSystem.build(const KernelSnapshot(), androidEnvironment);
-
-    expect(result.exceptions.values.single.exception, isA<Exception>());
+  test('KernelSnapshot handles null result from kernel compilation', () => testbed.run(() async {
+    final String build = androidEnvironment.buildDir.path;
+    processManager = FakeProcessManager.list(<FakeCommand>[
+      FakeCommand(command: <String>[
+        'bin/cache/dart-sdk/bin/dart',
+        'bin/cache/artifacts/engine/darwin-x64/frontend_server.dart.snapshot',
+        '--sdk-root',
+        'bin/cache/artifacts/engine/common/flutter_patched_sdk/',
+        '--target=flutter',
+        '-Ddart.developer.causal_async_stacks=false',
+        ...buildModeOptions(BuildMode.profile),
+        '--aot',
+        '--tfa',
+        '--packages',
+        '/.packages',
+        '--output-dill',
+        '$build/app.dill',
+        '--depfile',
+        '$build/kernel_snapshot.d',
+        '/lib/main.dart',
+      ], exitCode: 1),
+    ]);
+
+    expect(const KernelSnapshot().build(androidEnvironment), throwsA(isA<Exception>()));
   }));
 
-  test('kernel_snapshot does not use track widget creation on profile builds', () => testbed.run(() async {
-    final MockKernelCompiler mockKernelCompiler = MockKernelCompiler();
-    when(kernelCompilerFactory.create(any)).thenAnswer((Invocation _) async {
-      return mockKernelCompiler;
-    });
-    when(mockKernelCompiler.compile(
-      sdkRoot: anyNamed('sdkRoot'),
-      aot: anyNamed('aot'),
-      buildMode: anyNamed('buildMode'),
-      trackWidgetCreation: false,
-      targetModel: anyNamed('targetModel'),
-      outputFilePath: anyNamed('outputFilePath'),
-      depFilePath: anyNamed('depFilePath'),
-      packagesPath: anyNamed('packagesPath'),
-      mainPath: anyNamed('mainPath'),
-      extraFrontEndOptions: anyNamed('extraFrontEndOptions'),
-      fileSystemRoots: anyNamed('fileSystemRoots'),
-      fileSystemScheme: anyNamed('fileSystemScheme'),
-      linkPlatformKernelIn: anyNamed('linkPlatformKernelIn'),
-      dartDefines: anyNamed('dartDefines'),
-    )).thenAnswer((Invocation _) async {
-      return const CompilerOutput('example', 0, <Uri>[]);
-    });
+  test('KernelSnapshot does not use track widget creation on profile builds', () => testbed.run(() async {
+    globals.fs.file('.packages').writeAsStringSync('\n');
+    final String build = androidEnvironment.buildDir.path;
+    processManager = FakeProcessManager.list(<FakeCommand>[
+      FakeCommand(command: <String>[
+        'bin/cache/dart-sdk/bin/dart',
```

Comment:
Is there any way to query the path using the artifact API instead of hardcoding the paths?

Reply:
Yeah, I can use the artifact path API instead. That should be nicer

Reply:
Updated to use artifacts